packages/poller/src/invoice/processInvoices.ts

```diff
@@ -14,75 +13,397 @@ import {
+      logger.info('XERC20 strategy enabled for invoice destination, skipping', {
+        requestId, invoiceId, destinations: invoice.destinations, invoice, ticker: invoice.ticker_hash,
+        duration: getTimeSeconds() - start,
+      });
+      prometheus.recordInvalidPurchase(InvalidPurchaseReasons.DestinationXerc20, labels);
+      continue;
+    }
+
+    // Get the minimum amounts for invoice
+    const minAmounts = await (async () => {
+      try {
+        const { minAmounts: _minAmounts } = await everclear.getMinAmounts(invoiceId);
+        logger.debug('Got minimum amounts for invoice', {
+          requestId, invoiceId, invoice, minAmounts: _minAmounts, duration: getTimeSeconds() - start,
+        });
+        return _minAmounts;
+      } catch (e) {
+        logger.error('Failed to get min amounts for invoice', {
+          requestId, invoiceId, invoice, error: jsonifyError(e), duration: getTimeSeconds() - start,
+        });
+        return Object.fromEntries(invoice.destinations.map((d) => [d, '0']));
+      }
+    })();
+
+    // Set for ticker-destination pair lookup
+    const existingDestinations = new Set(
+      pendingPurchases.filter((p) => p.target.ticker_hash === invoice.ticker_hash).map((p) => p.purchase.params.origin),
+    );
+
+    // Filter out origins that already have pending purchases for this destination-ticker combo
+    let filteredMinAmounts = Object.fromEntries(
+      Object.entries(minAmounts).filter(([destination]) => {
+        if (existingDestinations.has(destination)) {
+          logger.info('Action exists for destination-ticker combo, removing from consideration', {
+            requestId, invoiceId, destination, duration: getTimeSeconds() - start,
+          });
+          prometheus.recordInvalidPurchase(InvalidPurchaseReasons.PendingPurchaseRecord, { ...labels, destination });
+          return false;
+        }
+        return true;
+      })
+    );
+
+    // Skip if no valid origins remain
+    if (Object.keys(filteredMinAmounts).length === 0) {
+      logger.info('No valid origins remain after filtering existing purchases', {
+        requestId, invoiceId, duration: getTimeSeconds() - start,
+      });
+      continue;
+    }
+
+    // Use all candidate origins in split calc for the first invoice of this ticker.
+    // For subsequent invoices, only use the chosen origin.
+    filteredMinAmounts = batchedGroup.origin ? 
+      { [batchedGroup.origin]: filteredMinAmounts[batchedGroup.origin] || '0' } : 
+      filteredMinAmounts;
+
+    const { intents, originDomain, totalAllocated } = await calculateSplitIntents(
+      invoice,
+      filteredMinAmounts,
+      config,
+      remainingBalances,
+      remainingCustodied,
+      logger,
+      requestId,
+    );
+
+    // Oldest invoice, prio enabled, and couldn't find a valid allocation
+    if (!originDomain && batchedGroup.invoicesWithIntents.length === 0 && config.prioritizeOldestInvoice) {
```

Comment:
if `config.prioritizeOldestInvoice` is false, then this `if` statement would get skipped.

the code would follow through to the next `if` and, assuming its purchasable, purchase the oldest invoice first because the invoices are ordered. 

instead should we pass the config flag into the grouping function, and allow it to sort the invoices to be processed that way?

Reply:
If disabled it goes to the next `if` - any purchaseable invoices were ID'd (whether oldest or not) and were split-cal'd oldest first. Ultimately it's always oldest first but the flag just forces oldest or nothing. 

What's the benefit of passing flag into the grouping function then?

Reply:
You could switch between oldest or newest first just in the sorting and keep the rest of the logic the same. Or am I missing something

Reply:
Oh it's meant to be a flag for forcing 'oldest or nothing' rather than the order. We always look by oldest-first. Unless newest-first is something we might want?

Reply:
ah maybe it should be renamed for clarity -- like force oldest vs oldest first